src/concern/test.rs

```diff
@@ -179,8 +179,9 @@ async fn snapshot_read_concern() {
         .database(function_name!())
         .collection::<Document>(function_name!());
 
-    // TODO RUST-734: Run this test on sharded clusters when transactions are complete.
-    if client.is_replica_set() && client.server_version_gte(4, 0) {
+    if client.is_replica_set() && client.server_version_gte(4, 0)
```

Comment:
Can we add a `supports_transactions` method to `TestClient` to perform these checks and call that instead in the places where we have this logic?